snooty/language_server.py

```diff
@@ -429,7 +429,16 @@ def m_text_document__resolve(
             )
             return str(resolved_target_path)
         elif resolveType == "directive":
-            return str(self.project.config.source_path) + fileName
+            """If the filename has a .rst extension, it might be converted from
+            a YAML file. We want to get its original file path in that case."""
+            if ".rst" in fileName:
+                filePath = str(self.project.config.source_path) + fileName
+                fileId = self.project.get_fileid(Path(filePath))
+                finalFileId = self.project.config.source_path / fileId
```

Comment:
This should be named something like `real_file_path`, since it's a path not a FileId instance